signer-service/src/api/services/siwe.service.js

```diff
@@ -30,29 +34,34 @@ const verifySiweMessage = async (nonce, signature, initialSiweMessage) => {
 
   const siweMessage = existingSiweDataForNonce.siweMessage
     ? existingSiweDataForNonce.siweMessage
-    : new siwe.SiweMessage(initialSiweMessage);
+    : SignInMessage.fromMessage(initialSiweMessage);
   const address = existingSiweDataForNonce.address;
 
   if (!siweMessage) {
     throw new Error('Message must be provided as a parameter if it has not been initially validated.');
   }
 
-  // Verify the integrity of the message
-  const publicClient = createPublicClient({
-    chain: polygon,
-    transport: http(),
-  });
-
-  const valid = await publicClient.verifyMessage({
-    address: address,
-    message: siweMessage.toMessage(), // Validation must be done on the message as string
-    signature,
-  });
+  // verify with substrate (generic) or evm generic (using polygon public client)
+  let valid = false;
+  if (address.startsWith('0x')) {
+    const publicClient = createPublicClient({
+      chain: polygon,
+      transport: http(),
+    });
+    valid = await publicClient.verifyMessage({
+      address: address,
+      message: siweMessage.toMessage(), // Validation must be done on the message as string
+      signature,
+    });
+  } else if (address.startsWith('5')) {
```

Comment:
Might not be guaranteed that every valid address starts with a 5. I think there could be some edgy edge cases where it doesn't. I could be wrong though. Let's rather try to call some decode function on the keyring and if there is some result, we assume it's a substrate address?

Reply:
I think the "generic substrate" format always [start with](https://wiki.polkadot.network/docs/learn-account-advanced) `5`. 
We can add extra verification but checking the checksum and ensure is at least valid for each option, but the success still depends on the UI sending the correct address that matches with one of the auth credentials, so I don't know if it adds much value, wdyt? It would mostly be for us to get better error if we send an invalid address.

Reply:
Right. Are we always sending the generic substrate address though? I guess you are right and we are. Would be important to keep doing it like that but I'm fine with keeping it like this then.